### PR TITLE
Review modal

### PR DESCRIPTION
**Pull request by chrsbell**
Basically complete

### CODE REVIEW

File: client/src/components/AppModal.jsx [hunk 1]
```diff
@@ -1,74 +1,121 @@
...
-import fonts from './Fonts.js';
+import {Fonts, FlexColumn, Container, animation} from './Constants.jsx';
 
-const Container = styled.div`
-background-color: white;
-border-style: solid;
-border-width: 5px;
-border-radius: 30px;
-margin: 3vh 10vw;
+const ReviewModal = styled.div.attrs(props => {
```
Comment (paprikatie) [inline client/src/components/AppModal.jsx:8]: can just use ( ) instead of { return { } }

File: client/src/components/BarGraph.jsx [hunk 2]
```diff
@@ -1,58 +1,79 @@
...
+}
+
+}
 `;
 
-const BarGraph = ({text, rating}) => {
-  const maxWidth = margins.barWidth; // width in px for higher accuracy
+const BarGraph = ({text, rating, isModal}) => {
+  const maxWidth = isModal ? margins.modalBarWidth : margins.barWidth; // width in px for higher accuracy
+  const rightMargin = isModal ? `10vw` : `0`;
```
Comment (paprikatie) [inline client/src/components/BarGraph.jsx:64]: Loving the ternaries

File: client/src/components/Constants.jsx [hunk 3]
```diff
@@ -1,28 +1,94 @@
...
+
+&.left {
+  justify-content: flex-start;
+  margin: 0 5vw;
+}
+
+`;
+
+const Text = styled.p.attrs(props => {
+  return {
```
Comment (paprikatie) [inline client/src/components/Constants.jsx:51]: same thing here, can use ( ) instead of { return { } }

File: client/src/components/Constants.jsx [hunk 4]
```diff
@@ -1,28 +1,94 @@
...
+
+&.rating {
+  font-size ${Fonts.small};
+  padding-left: 10px;
+}
 `;
 
-const FlexColumn = styled.div`
+const FlexColumn = styled.div.attrs(props => {
+  return {
```
Comment (paprikatie) [inline client/src/components/Constants.jsx:68]: ''

File: client/src/components/Constants.jsx [hunk 5]
```diff
@@ -1,28 +1,94 @@
...
+margin-left: 1vw;
+&.modal {
+  margin-left: -20px;
+  margin-top: -50px;
+}
 `;
 
-const Container = styled.div`
+const Container = styled.div.attrs(props => {
+  return {
```
Comment (paprikatie) [inline client/src/components/Constants.jsx:82]: "

File: client/src/components/ReviewListModal.jsx [hunk 6]
```diff
@@ -1,38 +1,44 @@
...
 
-const FlexRow = styled.div`
-display: flex;
-flex-direction: row;
-max-width: 2vw;
-`;
-
-const FlexColumn = styled.div`
+const ScrollableFlexColumn = styled.div.attrs(props => {
+  return {
```
Comment (paprikatie) [inline client/src/components/ReviewListModal.jsx:9]: ''

File: client/src/components/ReviewListModal.jsx [hunk 7]
```diff
@@ -43,58 +49,104 @@ class ReviewListModal extends React.Component {
...
+      })
+    });
+  }
+
   render() {
+    const areViewableReviews = this.state.viewableReviews && this.state.viewableReviews.length;
+    let reviewComponents = null;
+    if (areViewableReviews) {
+      reviewComponents = this.state.viewableReviews.map((review, i) => {
+        return (
```
Comment (paprikatie) [inline client/src/components/ReviewListModal.jsx:130]: ''

File: client/src/components/SearchBar.jsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+align-items: center;
+padding: 12px 12px 12px 16px;
+border-radius: 100px;
+box-shadow: rgb(176, 176, 176) 0px 0px 0px 1px inset;
+`;
+
+class SearchBar extends React.Component {
+  constructor({callback}) {
+    super();
+    this.search = callback;
```
Comment (paprikatie) [inline client/src/components/SearchBar.jsx:45]: could be a stateless component! no state

**Review by paprikatie — CHANGES_REQUESTED** [review]
Optimizing tweaks for usage of ( ) in many places instead of { return { } }
and could change component to stateless when it doesn't use any state properties!

File: client/src/components/BarGraph.jsx (hunk 2 above)
Comment (chrsbell) [reply client/src/components/BarGraph.jsx:64]: lets goo

File: client/src/components/AppModal.jsx (hunk 1 above)
Comment (chrsbell) [reply client/src/components/AppModal.jsx:8]: nice catch, changing to this

File: client/src/components/Constants.jsx (hunk 3 above)
Comment (chrsbell) [reply client/src/components/Constants.jsx:51]: ^

File: client/src/components/Constants.jsx (hunk 4 above)
Comment (chrsbell) [reply client/src/components/Constants.jsx:68]: ^

File: client/src/components/AppModal.jsx (hunk 1 above)
Comment (chrsbell) [reply client/src/components/AppModal.jsx:8]: ^

File: client/src/components/SearchBar.jsx (hunk 8 above)
Comment (chrsbell) [reply client/src/components/SearchBar.jsx:45]: Added state to my search bar since making this branch but nice suggestion

**Pull request merged**